### PR TITLE
[Mobile Payments] JITMs action should be able to handle universal links

### LINKED ISSUE #8460: [Mobile Payments] JITMs action should be able to handle universal links

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #8460

### Description

The PR  makes JITM more versatile, so it now handles the given link as a universal link. So if link is supported by the app then specific screen in the app is open, if not then browser is open

### Testing instructions

* Receive JITM IPP
* Click on "purchase card reader"
* Notice that Web view is opened

Do the same but alter the URL returned from the [JITM endpoint](path=%2Fjetpack%2Fv4%2Fjitm%2F%26_method%3Dget&json=true&query=%7B%22message_path%22%3A%22woomobile%3Amy_store%3A) to `https://woocommerce.com/mobile/payments/tap-to-pay`  using code change or proxy change the URL. Notice that the "TTP summary screen" is opened

### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AnirudhBhat** [comment]
I haven't started the review yet. I observed an app crash as soon as I clicked on the "Purchase a card reader" button in the JITM with the following error message:

`Calling startActivity() from outside of an Activity  context requires the  
FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?`

Probably because the `JitmClickHandler` uses the `application context` and not the `activity` one.

**Comment by kidinov** [comment]
@AnirudhBhat good catch! I fixed that - a981a43905611e995e212445872e72d684a838c1

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -290,8 +291,10 @@ class MainActivity :
             checkForAppUpdates()
         }
 
-        viewModel.handleIncomingAppLink(intent?.data)
-        viewModel.handleShortcutAction(intent?.action?.toLowerCase())
+        if (savedInstanceState == null) {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:294]: We were opening the same intent when the activity was recreated, e.g. after the config change. Do you see any side effects from this change?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt [hunk 2]
```diff
@@ -122,6 +122,9 @@ class MainActivityViewModel @Inject constructor(
             ResolveAppLink.Action.ViewTapToPay -> {
                 triggerEvent(ViewTapToPay)
             }
+            is ResolveAppLink.Action.ViewUrlInWebView -> {
+                triggerEvent(ViewUrlInWebView(event.url))
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt:126]: ❓ I see a test is missing for this logic. What do you think about adding one?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt:294]: This should be fine 👍

**Review by AnirudhBhat — COMMENTED** [review]
Thanks for your work on this @kidinov I've tested the changes and the functionality itself works fine. I've found one issue related to the back stack when the deep link is triggered. It can be understood best by seeing the video below.


https://user-images.githubusercontent.com/1331230/227129654-efa667bc-5929-4eab-8477-28c88251bf71.mp4

The same behavior can be observed when opening a URL in webview as well since this is also being handled as a deep link.

I believe what's happening is that when the deep link is triggered into our app, the start destination from each `navigation` element in the hierarchy is also added to the stack.

>When [nesting graphs](https://developer.android.com/guide/navigation/navigation-nested-graphs), the start destination from each level of nesting—that is, the start destination from each <navigation> element in the hierarchy—is also added to the stack. This means that when a user presses the Back button from a deep link destination, they navigate back up the navigation stack just as though they entered your app from its entry point.
-[Documentation](https://developer.android.com/guide/navigation/navigation-deep-link)

We could probably overcome this by adding `FLAG_ACTIVITY_NEW_TASK` in conjunction with `FLAG_ACTIVITY_CLEAR_TASK`. WDYT?

**Comment by kidinov** [comment]
@AnirudhBhat thank you, good catch. Fixed as advised here - 7751b508a47691ef4b5e3985b9ebc29d80c4c840

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivityViewModel.kt:126]: Done - 57401543831a29e6d00d4550757f54f39d81d20e

**Comment by AnirudhBhat** [comment]
LGTM! :shipit:

**Review by AnirudhBhat — APPROVED** [review]

**Pull request merged**